src/Mail.java

```diff
@@ -0,0 +1,17 @@
+import java.util.Date;
+
+public abstract class Mail {
+    protected Type type;
```

Comment:
this need a constructor, so you can enforce and ensure all of these fields are working in all child classes